pulp_ansible/app/serializers.py

```diff
@@ -270,17 +271,65 @@ class CollectionVersionSerializer(SingleArtifactContentSerializer, ContentChecks
         help_text=_("The namespace of the collection."), max_length=32
     )
 
-    repository = serializers.URLField(
-        help_text=_("The URL of the originating SCM repository."), allow_blank=True, max_length=128
-    )
-
     tags = TagNestedSerializer(many=True, read_only=True)
 
     version = serializers.CharField(help_text=_("The version of the collection."), max_length=32)
 
+    def validate(self, data):
```

Comment:
Please consider performing that validation work in `deferred_validate`.

See: https://pulp.plan.io/issues/5403?pn=1#note-9